kitsune/wiki/utils.py

```diff
@@ -174,12 +174,15 @@ def get_featured_articles(
             product=product,
             locale=locale,
             fetch_for_aaq=fetch_for_aaq,
-        )[:limit]
+        )
     )
 
-    if len(pinned_articles) == limit:
+    if (num_pinned_articles := len(pinned_articles)) == limit:
         return pinned_articles
 
+    elif num_pinned_articles > limit:
```

Comment:
nit - this could be reduced to:
```python
if len(pinned_articles) <= limit:
    return pinned_articles
return random.sample(pinned_articles, limit)
```